.github/workflows/playwright-changed-files.yml

```diff
@@ -0,0 +1,83 @@
+name: Playwright E2E Tests - Changed Files
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened]
+
+# Avoid duplicate workflows on same branch
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}-playwright-changed-files
+  cancel-in-progress: true
+
+jobs:
+  test:
+    runs-on: ubuntu-latest-8-cores
+
+    defaults:
+      run:
+        shell: bash --login -eo pipefail {0}
+
+    steps:
+      - name: Checkout Streamlit code
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ inputs.ref }}
+          persist-credentials: false
+          submodules: "recursive"
+          fetch-depth: 2
+      - name: Get changed files
+        id: changed-files
+        uses: tj-actions/changed-files@v44
+        with:
+          path: e2e_playwright
+          files: "**/*_test.py"
+      - name: Check changed files
+        id: check_changed_files
+        env:
+          CHANGED_FILES: ${{ steps.changed-files.outputs.all_changed_and_modified_files }}
+          CHANGED_FILES_COUNT: ${{ steps.changed-files.outputs.all_changed_and_modified_files_count }}
+        run: |
+          echo "Changed files count: ${CHANGED_FILES_COUNT}"
+          echo "$CHANGED_FILES"
+          if [[ "${CHANGED_FILES_COUNT}" -gt 5 || "${CHANGED_FILES_COUNT}" -lt 1 ]]; then
+            echo "This workflow only supports between 1-5 changed files. Otherwise its skipping running the tests.";
```

Comment:
can you add a comment / extend this about the *why*? I assume its because if there are more than X (5) tests, there is no advantage to run this playwright-flow in addition to the standard playwright flow we have that executes all tests anyways

Reply:
Yep, I added this as a comment.